src/main/java/com/uber/cadence/testing/WorkflowShadower.java

```diff
@@ -114,6 +116,28 @@ public void run() throws Throwable {
     } while (nextPageToken != null && options.getShadowMode() == Mode.Normal);
   }
 
+  public void registerWorkflowImplementationTypes(Class<?>... workflowImplementationClasses) {
```

Comment:
I wonder how does the test work if we can't register workflow to the shadower?

Reply:
In the test, I register workflows in the replay activity, then inject the replay activity to the shadower. Users doesn't need to know the activity.